src/reconcile/rec_write.c

```diff
@@ -2350,39 +2398,132 @@ __rec_split_grow(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t add_len)
 {
 	WT_BM *bm;
 	WT_BTREE *btree;
-	size_t corrected_page_size, len;
+	size_t corrected_page_size, inuse, len;
 
 	btree = S2BT(session);
 	bm = btree->bm;
 
 	len = WT_PTRDIFF(r->first_free, r->disk_image.mem);
-	corrected_page_size = len + add_len;
+	inuse = len - r->bnd[r->bnd_next].offset +
```

Comment:
Line wrapping isn't quite right here, and the block is a bit hard to read (but that's my personal preference)